pkg/occlient/occlient.go

```diff
@@ -1328,31 +1338,127 @@ func (c *Client) GetClusterServiceClasses() ([]scv1beta1.ClusterServiceClass, er
 }
 
 // CreateServiceInstance creates service instance from service catalog
-func (c *Client) CreateServiceInstance(componentName string, componentType string, labels map[string]string) error {
-	// Creating Service Instance
-	_, err := c.serviceCatalogClient.ServiceInstances(c.namespace).Create(
+func (c *Client) CreateServiceInstance(serviceName string, serviceType string, servicePlan string, parameters map[string]string, labels map[string]string) error {
+	serviceInstanceParameters, err := serviceInstanceParameters(parameters)
+	if err != nil {
+		return errors.Wrapf(err, "unable to create the service instance parameters")
+	}
+
+	_, err = c.serviceCatalogClient.ServiceInstances(c.namespace).Create(
 		&scv1beta1.ServiceInstance{
 			TypeMeta: metav1.TypeMeta{
 				Kind:       "ServiceInstance",
 				APIVersion: "servicecatalog.k8s.io/v1beta1",
 			},
 			ObjectMeta: metav1.ObjectMeta{
-				Finalizers: []string{"kubernetes-incubator/service-catalog"},
-				Name:       componentName,
-				Namespace:  c.namespace,
-				Labels:     labels,
+				Name:      serviceName,
+				Namespace: c.namespace,
+				Labels:    labels,
 			},
 			Spec: scv1beta1.ServiceInstanceSpec{
 				PlanReference: scv1beta1.PlanReference{
-					ClusterServiceClassExternalName: componentType,
+					ClusterServiceClassExternalName: serviceType,
+					ClusterServicePlanExternalName:  servicePlan,
 				},
+				Parameters: serviceInstanceParameters,
 			},
-			Status: scv1beta1.ServiceInstanceStatus{},
 		})
 
 	if err != nil {
-		return errors.Wrap(err, "unable to create service instance")
+		return errors.Wrapf(err, "unable to create the service instance %s for the service type %s and plan %s", serviceName, serviceType, servicePlan)
 	}
+
+	// Create the secret containing the parameters of the plan selected.
+	err = c.CreateServiceBinding(serviceName, c.namespace, parameters)
+	if err != nil {
+		return errors.Wrapf(err, "unable to create the secret %s for the service instance", serviceName)
+	}
+
+	return nil
+}
+
+// CreateServiceBinding creates a ServiceBinding (essentially a secret) within the namespace of the
+// service instance created using the service's parameters.
+func (c *Client) CreateServiceBinding(componentName string, namespace string, parameters map[string]string) error {
+	serviceInstanceParameters, err := serviceInstanceParameters(parameters)
+	if err != nil {
+		return errors.Wrapf(err, "unable to create the service instance parameters")
```

Comment:
Should use `Wrap` since no parameters are passed.

Reply:
Good catch, will fix